nerfstudio/data/datamanagers/parallel_datamanager.py

```diff
@@ -0,0 +1,220 @@
+            self.config.dataparser.data = Path(self.config.data)
+        else:
+            self.config.data = self.config.dataparser.data
+        self.dataparser = self.dataparser_config.setup()
+        if test_mode == "inference":
+            self.dataparser.downscale_factor = 1  # Avoid opening images
+        self.includes_time = self.dataparser.includes_time
+        self.train_dataparser_outputs: DataparserOutputs = self.dataparser.get_dataparser_outputs(split="train")
+        self.train_dataset = self.create_train_dataset()
+        # self.eval_dataset = self.create_eval_dataset()
+        self.pix_sampler = self._get_pixel_sampler(self.train_dataset, self.config.train_num_rays_per_batch)
+        # Spawn is critical for not freezing the program (PyTorch compatability issue)
+        mp.set_start_method("spawn")
+        # Manager().queue() can be faster
+        # https://stackoverflow.com/questions/43439194/python-multiprocessing-queue-vs-multiprocessing-manager-queue/45236748#45236748
+        self.data_q = mp.Manager().Queue(maxsize=2)
+        self.data_procs = [
+            DataProc(self.data_q, self.config, self.train_dataparser_outputs, self.train_dataset, self.pix_sampler)
+            for i in range(self.config.n_procs)
+        ]
+        for proc in self.data_procs:
+            proc.start()
+        super().__init__()
+        # Prime the executor with the first batch
+        self.executor = concurrent.futures.ThreadPoolExecutor(max_workers=1)
+        self.batch_fut = self.executor.submit(self.data_q.get)
+
+    def create_train_dataset(self) -> TDataset:
+        """Sets up the data loaders for training"""
+        return self.dataset_type(
+            dataparser_outputs=self.train_dataparser_outputs,
+            scale_factor=self.config.camera_res_scale_factor,
+        )
+
+    def _get_pixel_sampler(self, dataset: TDataset, *args: Any, **kwargs: Any) -> PixelSampler:
+        """Infer pixel sampler to use."""
+        if self.config.patch_size > 1:
+            return PatchPixelSampler(*args, **kwargs, patch_size=self.config.patch_size)
+
+        # If all images are equirectangular, use equirectangular pixel sampler
+        is_equirectangular = dataset.cameras.camera_type == CameraType.EQUIRECTANGULAR.value
+        if is_equirectangular.all():
+            return EquirectangularPixelSampler(*args, **kwargs)
+        # Otherwise, use the default pixel sampler
+        if is_equirectangular.any():
+            CONSOLE.print("[bold yellow]Warning: Some cameras are equirectangular, but using default pixel sampler.")
+        return PixelSampler(*args, **kwargs)
+
+    def next_train(self, step: int) -> Tuple[RayBundle, Dict]:
+        self.train_count += 1
+
+        # Fetch the next batch in an executor to parallelize the queue get() operation
+        # with the train step
+        bundle, batch = self.batch_fut.result()
+        self.batch_fut = self.executor.submit(self.data_q.get)
```

Comment:
will this work for multi-gpu distributed data parallel (ddp) mode?  (do you need multi-gpu testing support?)   

can this class now just be a torch DataLoader, which I think would automatically support ddp?  i guess i thought the original `DataManager` was not a DataLoader and is a `nn.Module` to afford for camera optimizer / gradients